### PR TITLE
add game summary and refactor game controller

### PR DESCRIPTION
**Pull request by ghemDD**

### CODE REVIEW

**Review by JorisMonnet — APPROVED** [review]
Some dependencies could be updated
Token spotify is clear, this is really dangerous !

File: app/src/androidTest/java/ch/epfl/sdp/blindwar/tutorial/DemoActivityTest.kt [hunk 1]
```diff
@@ -21,44 +28,112 @@ class DemoActivityTest {
         DemoActivity::class.java
     )
 
-    @Test
-    fun isPlayClickable() {
-        onView(withId(R.id.startButton)).check(matches(isClickable()))
+    private val longString = "The mare flattened her ears against her skull and snorted throwing up earth with her hooves she didnt want to go" +
```
Comment (marluxiaboss) [inline app/src/androidTest/java/ch/epfl/sdp/blindwar/tutorial/DemoActivityTest.kt:31]: This name of variable is not very well chosen.

**Comment by JorisMonnet** [comment]
I approved because token has been modified

File: app/src/androidTest/java/ch/epfl/sdp/blindwar/tutorial/DemoActivityTest.kt [hunk 2]
```diff
@@ -21,44 +28,112 @@ class DemoActivityTest {
...
+    /**
+    /** 30 seconds to guess **/
     @Test
-    fun guessResetAfterClick() {
+    fun timeOutTest() {
         onView(withId(R.id.guessEditText))
-            .perform(clearText(), typeText("THIS IS NOT CORRECT"))
-        onView(withId(R.id.guessButton)).perform(closeSoftKeyboard()).perform(click())
-        onView(withId(R.id.guessEditText)).check(matches(withText("")))
+            .perform(clearText(), typeText(longString))
```
Comment (marluxiaboss) [inline app/src/androidTest/java/ch/epfl/sdp/blindwar/tutorial/DemoActivityTest.kt:90]: There are a lot of commented sections like this that should be deleted when not needed, it makes the code a bit messy.

**Review by marluxiaboss — APPROVED** [review]
You should add a bit of explanatory comments

File: app/src/androidTest/java/ch/epfl/sdp/blindwar/tutorial/DemoActivityTest.kt (hunk 1 above)
Comment (CHOCOLVT) [inline app/src/androidTest/java/ch/epfl/sdp/blindwar/tutorial/DemoActivityTest.kt:31]: never used, was here for timeout but remove here, to adapt

**Review by CHOCOLVT — APPROVED** [review]
more doc would have been great,
but generally great code

**Pull request merged**